### PR TITLE
[WIP] Add support for multi device.

### PR DESCRIPTION
**Pull request by AAbouZaid**
Thanks for your work, it's really helpful.
As you know, since it's still on going, there are many things are hard-coded in VM devices like network and hard disks.

So this PR is to make it more dynamic specially to define more than 1 net and hd device.
And all changes are taking backward comparability in mind.

The work still in progress, and I will let you know once I'm done.
But feel free to mention anything you see could be done better.
I came from Python background, and my golang knowledge is modest.

I also will open [another PR](https://github.com/Telmate/terraform-provider-proxmox/pull/13) for terraform-provider-proxmox.

Thanks.

### CODE REVIEW

File: proxmox/config_qemu.go [hunk 1]
```diff
@@ -11,28 +12,30 @@ import (
 	"time"
 )
 
+type qemuSets []map[int]interface{}
```
Comment (aabouzaid) [inline proxmox/config_qemu.go]: This should be a map not slice. I will fix it.

**Comment by AAbouZaid** [comment]
@ggongaware 
I'm almost done and added all what I think it's necessary :tada: 
It now supports more than 1 disk and 1 nic, with more options for both of them.
And NICs have VLAN and MAC address as you mentioned.

Could you please check it and tell me if we need anything else (I think we need to take a look on `prepareDiskSize` for each disk).

Thanks

**Comment by ggongaware** [comment]
Thank you, I see the "mac=" part, but I didn't see a "vlan=" part of the netX strings.

**Comment by AAbouZaid** [comment]
@ggongaware I'm using the official naming `tag` as it used in Proxmox.

But unlike the MAC address, VLAN doesn't need to special handling.
Any sub config has `key=value` is the default style, but the MAC and model are stored as `model:mac` which is need special handling than rest of values.

I'm using the official API docs as a reference.
https://pve.proxmox.com/pve-docs/api-viewer/index.html

Thanks :-)

**Comment by AAbouZaid** [comment]
@ggongaware This PR is ready to be merged :tada:

**Comment by ggongaware** [comment]
I should run some tests on this branch, can you update ReadMe with a sample of the new json format?

**Comment by AAbouZaid** [comment]
@ggongaware Done :-)

Could you please make some extra tests for cloning? because I did tests mainly for create new VMs but not too much for clone.
Thanks.

**Comment by themouette** [comment]
@ggongaware is there any reason (except for the conflict) that prevent this from being merged?

**Comment by ggongaware** [comment]
I haven't had a chance to test it against a bunch of templates I need to maintain. Its got some definite improvements. I may end up cutting another branch for my templates as-is, and try to move forward with this.

**Comment by AAbouZaid** [comment]
I'd like to see it too :-)
Especially that backward compatibility is already in mind (just need someone to test).

**Comment by themouette** [comment]
Thank you for the quick answer! 🙇

**Comment by ggongaware** [comment]
@AAbouZaid if you can resolve the conflicts, I'll merge it into master

**Comment by AAbouZaid** [comment]
@ggongaware Could you please review the use of `sizeunit` package (i.e. where it fits with multi disks) because I'm not sure if I follow its logic.

Or you can give me an example and I will use it again.

**Pull request merged**